src/sdam/description/topology/server_selection/mod.rs

```diff
@@ -39,29 +105,17 @@ impl TopologyDescription {
         }
     }
 
-    pub(crate) fn select_server<'a>(
+    pub(crate) fn suitable_servers_in_latency_window<'a>(
         &'a self,
         criteria: &'a SelectionCriteria,
-    ) -> Result<Option<&'a ServerDescription>> {
+    ) -> Result<Vec<&'a ServerDescription>> {
         if let Some(message) = self.compatibility_error() {
             return Err(ErrorKind::ServerSelectionError {
                 message: message.to_string(),
             }
             .into());
         }
 
-        // Although `suitable_servers` has to handle the Unknown case for testing, we skip calling
```

Comment:
Given that the new functions above are not actually called by this method anywhere, why do we need to remove these optimizations? Returning an empty or singleton vector will still be correct in these cases, since we'll never have two servers to choose from.

Reply:
I removed the first one since it would no longer be avoiding an allocation, but it looks like we want to keep that one in since we don't want to bother with the predicate for an unknown topology. 

I removed the second one since we were skipping the predicate in that case which is technically a minor bug (a user could decide to fail server selection if the single server doesn't meet their criteria). Also, the logic we're skipping is the same that case for read preferences anyways. 

The last optimization (previously on line 74) is also a bug, since we still need to filter the mongoses based on latency even if the user specifies a Primary read preference.